### PR TITLE
Binary to unicode converter

### PR DESCRIPTION
**Pull request by zachthompson**
**What does your Instant Answer do?**

Converts binary into unicode characters much like the [Ascii to Binary IA](https://duck.co/ia/view/ascii).  However, it has some additional features:

1. Converts to unicode outside of the ascii range.
2. Will convert a list of binary strings separated by spaces.
3. Retains the original query's spacing.

**What problem does your Instant Answer solve (Why is it better than organic links)?**

Every now and again you have some funny person that encodes a sentence in binary and you want to quickly decode it.  Here's your IA.  I don't want to decode it letter by letter.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

Geeks and the curious.

**Which existing Instant Answers will this one supercede/overlap with?**

https://duck.co/ia/view/ascii  We could potentially decommission ascii.

---
IA Page: https://duck.co/ia/view/bin2unicode

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Bin2Unicode.pm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+use DDG::Goodie;
+use strict;
+
+no warnings 'non_unicode';
+
+zci answer_type => 'bin2unicode';
+zci is_cached   => 1;
+
+triggers query => qr{[ 01]+(?:(?:to\s+)?unicode|text|ascii)?};
```
Comment (mintsoft) [inline lib/DDG/Goodie/Bin2Unicode.pm]: couldn't this be `triggers end => qw(unicode text ascii)`

Comment (zachthompson) [inline lib/DDG/Goodie/Bin2Unicode.pm]: @mintsoft It could but those are all optional.  So "01001101 01110101 01100011 01101000" as well as "01001101 01110101 01100011 01101000 to text" will work.  It's an open question I guess if we want the former to trigger.  I think we do ;)

Comment (mintsoft) [inline lib/DDG/Goodie/Bin2Unicode.pm]: That's fair enough I can certainly see the use-case of the pure-binary triggering

Comment (moollaza) [inline lib/DDG/Goodie/Bin2Unicode.pm]: @zachthompson I think we'll want to anchor this regex? `201 to text` triggers this.

It may also help to add a capture to the regex and `handle matches` below?

**Comment by zachthompson** [comment]
Some fairly major changes:

* Cover some edge cases, e.g. erroneously trying to parse hex
* Default to ascii when exceeding the max code point.  This allows for long strings of multiple characters (no spaces) like the original.
* Tests

**Comment by zachthompson** [comment]
BTW, this should now cover all of the space that Ascii did.  So if this seems like a more robust route, we might as well remove it as part of the PR.

**Comment by moollaza** [comment]
@zachthompson this is looking good but I think we need to make sure the rendering unicode is visible? and exists:

1111_1111_to_unicode_at_duckduckgo_and_any_do
0000000000_to_unicode_at_duckduckgo_and_any_do

I'm just throwing 1's and 0's at it right now to see what happens...

File: t/Bin2Unicode.t [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+use strict;
+use warnings;
+
+zci answer_type => 'bin2unicode';
+zci is_cached   => 1;
+
+sub gen_struc_ans {
+    my ($in, $bin, $str, $asc) = @_;
+
+    return "Binary '$bin' converted to " . $asc ? 'ascii' : 'unicode' . " is '$str'",,
```
Comment (moollaza) [inline t/Bin2Unicode.t]: @zachthompson now that I see the double comma again `,,` is that intentional? (I just removed it in the PM, was pushing a commit to link the IA page and PR)

**Comment by zachthompson** [comment]
Give it a shot now. Comma was a typo.

**Comment by moollaza** [comment]
@zachthompson Thanks! This LGTM :+1:
